scrapely/_htmlpage.pyx

```diff
@@ -139,15 +139,14 @@ cdef class ScriptParser:
     cdef int parse(self, Py_UCS4 c, int i):
         if self.state == 10:
             self.state = 1
-
         if ((self.state == 1 and c == u'<') or
             (self.state == 2 and c == u'/') or
-            (self.state == 3 and c == u's') or
-            (self.state == 4 and c == u'c') or
-            (self.state == 5 and c == u'r') or
-            (self.state == 6 and c == u'i') or
-            (self.state == 7 and c == u'p') or
-            (self.state == 8 and c == u't') or
+            (self.state == 3 and c == u's' or c == u'S') or
```

Comment:
After doing some speed tests the difference is negligible.

I would be open to changing it to:
```python
        if ((self.state == 1 and c == u'<') or
            (self.state == 2 and c == u'/') or
            (self.state == 3 and c in u'sS') or
            (self.state == 4 and c in u'cC') or
            (self.state == 5 and c in u'rR') or
            (self.state == 6 and c in u'iI') or
            (self.state == 7 and c in u'pP') or
            (self.state == 8 and c in u'tT') or
            (self.state == 9 and c == u'>')):
            self.state += 1
```
Here's the generated C code:
```C
// Current
  __pyx_t_2 = ((__pyx_v_self->state == 3) != 0);
  if (!__pyx_t_2) {
    goto __pyx_L11_next_or;
  } else {
  }
  __pyx_t_2 = ((__pyx_v_c == 0x73) != 0);
  if (!__pyx_t_2) {
  } else {
    __pyx_t_1 = __pyx_t_2;
    goto __pyx_L5_bool_binop_done;
  }
  __pyx_L11_next_or:;
  __pyx_t_2 = ((__pyx_v_c == 83) != 0);
  if (!__pyx_t_2) {
  } else {
    __pyx_t_1 = __pyx_t_2;
    goto __pyx_L5_bool_binop_done;
  }


// Proposed (Same code generated when using u'sS' and [u's', u'S']
  __pyx_t_2 = ((__pyx_v_self->state == 3) != 0);
  if (!__pyx_t_2) {
    goto __pyx_L10_next_or;
  } else {
  }
  switch (__pyx_v_c) {
    case 0x73:
    case 83:
    __pyx_t_2 = 1;
    break;
    default:
    __pyx_t_2 = 0;
    break;
  }
  __pyx_t_3 = (__pyx_t_2 != 0);
  if (!__pyx_t_3) {
  } else {
    __pyx_t_1 = __pyx_t_3;
    goto __pyx_L5_bool_binop_done;
  }
```